### PR TITLE
Use ogdl-swift for Cartfile parsing

### PR DESCRIPTION
**Pull request by jspahrsummers**
_Depends on https://github.com/Carthage/ogdl-swift/pull/9._

This will refactor our `Cartfile` and `Cartfile.resolved` parsing logic to use [ogdl-swift](https://github.com/Carthage/ogdl-swift). No new features are planned for this PR.

/cc @robrix 

**To do:**

 - [x] Improve error messaging, where we have enough information to do so (i.e., excluding https://github.com/robrix/Madness/issues/47)
 - [x] Fix ridiculous symbol visibility

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -25,3 +25,18 @@
...
+	url = https://github.com/robrix/Box.git
+[submodule "Carthage/Checkouts/Prelude"]
+	path = Carthage/Checkouts/Prelude
+	url = https://github.com/robrix/Prelude.git
+[submodule "Carthage/Checkouts/Either"]
+	path = Carthage/Checkouts/Either
+	url = https://github.com/robrix/Either.git
+[submodule "Carthage/Checkouts/Madness"]
+	path = Carthage/Checkouts/Madness
+	url = https://github.com/robrix/Madness.git
```
Comment (robrix) [inline .gitmodules]: Incidentally, all of these are tagged for Swift 1.1 support, with 1.2b1+ on `master`. We can tag newer versions at any time so as not to hold up #326.

**Comment by jspahrsummers** [comment]
The fundamental approach here builds. Now I just need to figure out why I can't parse one of our test Cartfiles. :worried:

**Comment by jspahrsummers** [comment]
> Now I just need to figure out why this can't parse our test Cartfiles. :worried: 

:boom: https://github.com/Carthage/ogdl-swift/pull/9

**Comment by jspahrsummers** [comment]
Ready for review, though note the dependency.

**Comment by robrix** [comment]
Dependency merged.

**Comment by jspahrsummers** [comment]
Y
O
L
O

**Pull request merged**